ext/src/helpers/output_limited_buffer.rs

```diff
@@ -0,0 +1,50 @@
+use magnus::{value::InnerValue, value::Opaque, RString, Ruby};
+use std::io;
+
+pub struct OutputLimitedBuffer {
+    buffer: Opaque<RString>,
+    /// The maximum number of bytes that can be written to the output stream buffer.
+    capacity: usize,
+}
+
+impl OutputLimitedBuffer {
+    #[must_use]
+    pub fn new(buffer: Opaque<RString>, capacity: usize) -> Self {
```

Comment:
I'm not entirely sure if Ruby strings define a max length, but if it does should we validate that `capacity` is not greater than such length? Said differently: are we certain that an `RString` can hold `usize::MAX`?